ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import java.io.IOException;
+import org.apache.helix.HelixManager;
+import org.apache.helix.HelixManagerFactory;
+import org.apache.helix.InstanceType;
+import org.apache.helix.model.LeaderStandbySMD;
+import org.apache.helix.participant.StateMachineEngine;
+import org.json.JSONException;
+
+
+/**
+ * An implementation of {@link ClusterParticipant} that registers registers as a participant to a Helix cluster.
+ */
+class HelixParticipant implements ClusterParticipant {
```

Comment:
One question I am not completely sure about is, how a `HelixParticipant` participates in a cluster. The logic implemented in this class seem not to report changes at a granular basis. Maybe I missed some meaningful parts in the code, so I will come to you next week for more clarity.